### PR TITLE
Feature/ipaddressfunction

### PR DESCRIPTION
**Pull request by paucampana**
Add a function to get the next consecutive Ip.

### CODE REVIEW

File: src/foxops/engine/rendering.py [hunk 1]
```diff
@@ -14,6 +15,20 @@
...
 
+def first_ip_address_is_greater(first_ip, second_ip):
+    """Checks with of the two IP addresses is greater
+
+    Returns True if first one is greater. Otherwise False.
+    """
+    return ipaddress.ip_address(first_ip) > ipaddress.ip_address(second_ip)
+
+
+def get_next_ip(ip):
```
Comment (defreng) [inline src/foxops/engine/rendering.py:26]: should we make it more generic? I.e. call it `ip_add(ip: str, inc: int = 1)` and allow for arbitrary increases?

File: src/foxops/engine/rendering.py [hunk 2]
```diff
@@ -14,6 +15,20 @@
 logger = get_logger(__name__)
 
 
+def first_ip_address_is_greater(first_ip, second_ip):
```
Comment (defreng) [inline src/foxops/engine/rendering.py:18]: what do we need this for?

Also - if we need it, I would probably adapt name to make the usage look a bit more natural:

```
{{ ip | ip_address_is_greater_than(second_ip) }}
```

Comment (defreng) [inline src/foxops/engine/rendering.py:18]: I think we should move these filters into a separate file, i.e. `custom_filters.py` and also add some separate unit tests of these functions directly.

Comment (paucampana) [reply src/foxops/engine/rendering.py]: Name adapted. We want to be sure that we are providing a valid range of IPs. Otherwise we could have an invalid IPAddressPool.
possible error situations for our use case:
- range is provided in an incorrect order: 1.1.1.4-1.1.1.1
- range does not have at least 3 IPs: 1.1.1.1-1.1.1.2

File: tests/engine/test_custom_filters.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+from pathlib import Path
+
+from foxops.engine.rendering import create_template_environment, render_template_file
+
+
+async def test_rendering_custom_filter_increase_ip_add(tmp_path: Path):
```
Comment (defreng) [inline tests/engine/test_custom_filters.py:6]: actually, testing these custom filters we should do without the templating around...

we can just write some tests that directly call these functions

File: src/foxops/engine/custom_filters.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+def first_ip_address_is_greater_than(first_ip, second_ip):
+    """Checks with of the two IP addresses is greater
+
+    Returns True if first one is greater. Otherwise False.
+    """
+    return ipaddress.ip_address(first_ip) > ipaddress.ip_address(second_ip)
+
+
+def increase_ip_add(ip: str, inc: int = 1):
```
Comment (defreng) [inline src/foxops/engine/custom_filters.py:17]: I think we should have some consistency in the naming here - with a common prefix:

* `ip_is_greater_than`
* `ip_add`

Comment (paucampana) [reply src/foxops/engine/custom_filters.py]: changed

File: tests/engine/test_custom_filters.py (hunk 3 above)
Comment (paucampana) [reply tests/engine/test_custom_filters.py]: If we test directly the function we are not sure that these functions are available when we run the template. Apart from that we are calling functions in a different way (for example `ip_add_increase('1.1.1.1', 2)` vs `'1.1.1.1' | ip_add_increase(2)`). So I think it is a better test to check that is working inside a template.

File: src/foxops/engine/rendering.py (hunk 2 above)
Comment (paucampana) [reply src/foxops/engine/rendering.py]: As we agreed, I have removed this function for now.

File: tests/engine/test_custom_filters.py (hunk 3 above)
Comment (defreng) [reply tests/engine/test_custom_filters.py]: agreed!
Ideally - we should do both, verify that it's working in the template, but also that it covers all edge cases correctly.

I made some small adjustments to the tests - could you check out if that works for you? If yes, LGTM 🚀

**Comment by defreng** [comment]
@paucampana also, could you check your git configuration, Mr. FIRST_NAME LAST_NAME 😂
